assets/src/components/v2/departures/later_departures.tsx

```diff
@@ -0,0 +1,80 @@
+import React, {
+  useEffect,
+  useLayoutEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import _ from "lodash";
+import cx from "classnames";
+import weakKey from "weak-key";
+
+import { hasOverflowX } from "Util/util";
+
+import DepartureRow from "./departure_row";
+import { Row } from "./normal_section";
+import RoutePill from "./route_pill";
+
+const MAX_LATER_DEPARTURES = 5;
+
+const LaterDepatures = ({ rows }: { rows: Row[] }) => {
+  const [limit, setLimit] = useState(MAX_LATER_DEPARTURES);
+  const departures = useMemo(
+    () =>
+      _.take(
+        rows.filter(
+          (r): r is DepartureRow & { type: "departure_row" } =>
+            r.type === "departure_row",
+        ),
+        limit,
+      ),
+    [rows, limit],
+  );
+
+  const ref = useRef<HTMLDivElement>(null);
+  const [idx, setIdx] = useState(0);
+
+  useLayoutEffect(() => {
+    if (hasOverflowX(ref)) {
+      setLimit(Math.max(0, limit - 1));
+    }
+  }, [limit]);
+
+  useEffect(() => {
+    if (departures.length < 2) return;
+    const refreshMs = Math.min(3750, 15000 / departures.length);
+
+    const interval = setInterval(() => {
+      setIdx((i) => (i + 1) % departures.length);
+    }, refreshMs);
+
+    return () => clearInterval(interval);
+  }, [departures.length]);
+
+  const departure = departures[idx % departures.length];
+
+  return (
+    <div className="later-departures">
+      <div className="later-departures__header" ref={ref}>
+        <h3>Later Departures</h3>
+        <ol>
+          {departures.map((d) => (
+            <li
+              key={weakKey(d)}
```

Comment:
`DepartureRow`s have an `id` we could use.

Reply:
They do but `trimSections` "flattens" `DepartureRow`s with multiple times into multiple `DepartureRow`s reusing the same `id`. I do this to reduce layout shift when rotating through later departures. It does have the unfortunately mean that there can now be multiple `laterRows` with the same `id` ☹️

Reply:
Ah, right, makes sense. I think flattening the times is the intended behavior anyway (or at least how v1 behaves currently), otherwise Later Departures would consume even more vertical space, when the point of it is to save space.

This does seem like a case where someone (maybe future-us) could come across this, have the same thought I did, and "improve" it to use the `id`, unwittingly introducing a rendering bug. Maybe we could instead fix it by generating new IDs for the synthesized rows when we create them? Thus having basically the same effect as using `weakKey`, but preserving the contract that `id` is unique.

Reply:
Erm, I see you've pushed a change which now keeps times grouped when they're sent into later-departures, which is not the thing we want? Or at least, it's different from how v1 works, and I'm not sure we should change that in the absence of a design saying we should.

> I think flattening the times is the intended behavior anyway (or at least how v1 behaves currently)

Reply:
Oops! Easy enough to undo that. I had originally flattened the times because I ran into those layout issues I mentioned